### PR TITLE
Add support for PLAIN serialization

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: dialogue-example/src/main/java/com/palantir/dialogue/example/SampleServiceClient.java [hunk 1]
```diff
@@ -80,28 +84,31 @@ public HttpMethod httpMethod() {
...
 
             @Override
-            public String stringToString(String objectId, String header, String body) {
-                Preconditions.checkNotNull(objectId, "objectId parameter must not be null");
+            public SampleObject objectToObject(
+                    String path, OffsetDateTime header, List<ResourceIdentifier> query, SampleObject body) {
+                Preconditions.checkNotNull(path, "objectId parameter must not be null");
                 Preconditions.checkNotNull(header, "header parameter must not be null");
                 Preconditions.checkNotNull(body, "body parameter must not be null");
                 Request request = Request.builder()
```
Comment (uschi2000) [inline dialogue-example/src/main/java/com/palantir/dialogue/example/SampleServiceClient.java:100]: this is the interesting bit

**Comment by uschi2000** [comment]
Mark, I tagged the interesting bit.

File: dialogue-example/src/main/java/com/palantir/dialogue/example/SampleService.java [hunk 2]
```diff
@@ -16,8 +16,43 @@
...
+import com.palantir.ri.ResourceIdentifier;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+
 // Example of the interface code conjure would generate for a simple SampleService.
 public interface SampleService {
-    String stringToString(String objectId, String header, String body);
+
+    final class SampleObject {
```
Comment (markelliot) [inline dialogue-example/src/main/java/com/palantir/dialogue/example/SampleService.java]: Conjure probably doesn't end up nesting this object in generated code?

**Review by markelliot — APPROVED** [review]

File: dialogue-example/src/main/java/com/palantir/dialogue/example/SampleService.java (hunk 2 above)
Comment (uschi2000) [reply dialogue-example/src/main/java/com/palantir/dialogue/example/SampleService.java]: Correct. I'll move it up.

**Pull request merged**